<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integrated Framework to Annotate Genetic Variants • anor</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="life.css" rel="stylesheet">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- life --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="index.html">anor</a>
        <small class="life">developed by <a href="http://life2cloud.com">Jianfeng</a></small>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li>
  <a href="articles/introduction_to_anor.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/database_in_anor.html">Database in anor</a>
    </li>
    <li>
      <a href="articles/shiny_of_anor.html">Shiny app of anor</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">News</a>
</li>
        <li>
  <a href="https://github.com/JhuangLab/anor">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="anor" class="section level1">


<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The <a href="https://life2cloud.com/tools/anor">anor</a> package provides R functions as well as database resources which offer an integrated framework to annotate genetic variants from genome and transcriptome data. The wrapper functions of anor unified the interface of many published annotation tools, such as <a href="http://asia.ensembl.org/info/docs/tools/vep/index.html">VEP</a>, <a href="http://annovar.openbioinformatics.org/">ANNOVAR</a>, <a href="https://github.com/brentp/vcfanno">vcfanno</a> and <a href="http://www.bioconductor.org/packages/release/bioc/html/AnnotationDbi.html">AnnotationDbi</a>.</p>
<p>It also simplified the use of some of the external annotation tools in R. Besides, massive published genetic variants annotation databases were integrated into anor.</p>
<p>The main development motivation of anor is to increase the supported database and facilitate the variants annotation work. There are already too many tools and databases available and the usage is quite different. anor will integrate these annotation tools and get the merged annotation result in R.</p>
<p><img src="https://github.com/JhuangLab/anor/raw/master/man/figures/anor_package-1.jpg" align="center"></p>
<p>anor will not only provides annotation functions (both internal and external) but also hope to establish a shared annotation database resources. In fact, we have made some efforts in the relevant direction that massive scattered databases can easily to be download via using <a href="https://github.com/JhuangLab/BioInstaller">BioInstaller</a> R package.</p>
<p>Besides, we collected total 1,285 cases public B-progenitor acute lymphoblastic leukemia (B-ALL) transcriptome data from five different published datasets and built a novel large-scale transcript level sequencing variant database. <a href="https://software.broadinstitute.org/gatk/">The Genome Analysis Toolkit (GATK)</a>, <a href="http://massgenomics.org/varscan">VarScan2</a> and <a href="http://csb5.github.io/lofreq/">LoFreq</a> be used to call variants from the RNA-seq data (Database called BRVar). This work can help us to screen candidate systematic sequencing bias and evaluate variant calling trait from B-ALL RNA-seq.</p>
<p><img src="https://github.com/JhuangLab/anor/raw/master/man/figures/BRVar_databases-1.jpg" align="center"></p>
<p>If you want to download the 1,285 RNA-seq variants frequency database, you need click <a href="http://bioinfo.rjh.com.cn/labs/jhuang/contact.php">here</a> and send us a short application message (Whether it is for commercial use?). If the application is applied, we will reply you within 24 hours with a download required license code.</p>
<pre class="r"><span class="co"># Download BRVar database</span>
<span class="co"># You must input the applied license code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">anor</span>)
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="st">"db_annovar_brvar"</span>, <span class="st">"/path/annovar.dir"</span>,  <span class="kw">license</span> <span class="kw">=</span> <span class="st">"licence_code"</span>)</pre>
</div>
<div id="requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>anor annotation system:</p>
<ul>
<li>
<a href="https://cran.r-project.org/">R</a> &gt;= 3.3.0</li>
<li><a href="http://strawberryperl.com/">perl</a></li>
<li><a href="http://www.sqlite.org/download.html">SQLite</a></li>
<li><a href="http://www.bioconductor.org/packages/release/bioc/html/AnnotationDbi.html">AnnotationDbi</a></li>
</ul>
<p>ANNOVAR annotation system:</p>
<ul>
<li><a href="http://annovar.openbioinformatics.org/en/latest/">ANNOVAR</a></li>
</ul>
<p>VEP annotation system</p>
<ul>
<li><a href="http://strawberryperl.com/">perl</a></li>
<li><a href="http://asia.ensembl.org/info/docs/tools/vep/index.html">VEP</a></li>
</ul>
<p>vcfanno annotationo system</p>
<ul>
<li><a href="https://github.com/brentp/vcfanno">vcfanno</a></li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div id="core-softwares" class="section level3">
<h3 class="hasAnchor">
<a href="#core-softwares" class="anchor"></a>Core softwares</h3>
<pre class="r"><span class="co"># CRAN to install anor (The R package BioInstaller will be installed)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/setRepositories.html">setRepositories</a></span>(<span class="kw">ind</span><span class="kw">=</span><span class="fl">1</span>:<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">'anor'</span>)

<span class="co"># Github to install anor (The R package BioInstaller will be installed)</span>
<span class="co"># install.packages("devtools")</span>
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"JhuangLab/anor"</span>)

<span class="co"># Set needed directory</span>
<span class="no">annovar.dir</span> <span class="kw">&lt;-</span> <span class="st">"/opt/bin/annovar"</span>
<span class="no">database.dir</span> <span class="kw">&lt;-</span> <span class="st">"/opt/bin/annovar/humandb"</span>
<span class="no">vcfanno.dir</span> <span class="kw">&lt;-</span> <span class="st">"/opt/bin/vcfanno.dir"</span>

<span class="co"># Use BioInstaller to install ANNOVAR easily in R</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BioInstaller</span>)
<span class="fu"><a href="https://rdrr.io/pkg/BioInstaller/man/install.bioinfo.html">install.bioinfo</a></span>(<span class="st">'annovar'</span>, <span class="no">annovar.dir</span>)

<span class="co"># Use BioInstaller to install vcfanno easily in R</span>
<span class="co"># Support linux and mac</span>
<span class="fu"><a href="https://rdrr.io/pkg/BioInstaller/man/install.bioinfo.html">install.bioinfo</a></span>(<span class="st">'vcfanno'</span>, <span class="no">vcfanno.dir</span>)</pre>
</div>
<div id="annotation-database" class="section level3">
<h3 class="hasAnchor">
<a href="#annotation-database" class="anchor"></a>Annotation Database</h3>
<pre class="r"><span class="co"># Use download.database to download databases supported by anor and ANNOVAR</span>
<span class="co"># Some of examples as shown as below</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">anor</span>)
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="st">'db_annovar_refgene'</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"hg19"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="st">'db_ucsc_cytoband'</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"hg19"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="st">'db_annovar_avsnp147'</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"hg19"</span>)
<span class="co"># Or</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"db_annovar_refgene"</span>, <span class="st">"db_ucsc_cytoband"</span>),
  <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"hg19"</span>)

<span class="co"># All anor supported big annotation database required SQLite format</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="st">'db_annovar_avsnp147_sqlite'</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"hg19"</span>)</pre>
</div>
</div>
<div id="support-summary" class="section level2">
<h2 class="hasAnchor">
<a href="#support-summary" class="anchor"></a>Support Summary</h2>
<ul>
<li><a href="http://annovar.openbioinformatics.org/en/latest/">ANNOVAR databases</a></li>
<li>1285 cases B-ALL RNA-seq variants</li>
<li>Public RNA-editing databases</li>
<li>Other public database</li>
</ul>
</div>
<div id="basic-usage" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-usage" class="anchor"></a>Basic Usage</h2>
<pre class="r"><span class="co"># View the vignettes in anor package</span>
<span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes</a></span>(<span class="st">"anor"</span>)

<span class="co"># Get all anor supported annotation name</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/get.annotation.names.html">get.annotation.names</a></span>()

<span class="co"># Get annotation name needed download.name and </span>
<span class="co"># you can use download.database to download database using the download.name.</span>
<span class="no">download.name</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/get.download.name.html">get.download.name</a></span>(<span class="st">'avsnp147'</span>)

<span class="co"># Show download.name avaliable all versions database</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="kw">download.name</span> <span class="kw">=</span> <span class="no">download.name</span>, <span class="kw">show.all.versions</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co"># Download database in annotation database directory</span>
<span class="co"># Buildver default is hg19</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="kw">download.name</span> <span class="kw">=</span> <span class="no">download.name</span>, <span class="kw">version</span> <span class="kw">=</span> <span class="st">"avsnp147"</span>, <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"hg19"</span>,
  <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)

<span class="co"># Annotate variants from avsnp147 database use anor</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">data.table</span>)
<span class="no">chr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr1"</span>)
<span class="no">start</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"10020"</span>, <span class="st">"10020"</span>, <span class="st">"10020"</span>)
<span class="no">end</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"10020"</span>, <span class="st">"10020"</span>, <span class="st">"10020"</span>)
<span class="no">ref</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>)
<span class="no">alt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-"</span>, <span class="st">"-"</span>, <span class="st">"-"</span>)
<span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="kw">chr</span> <span class="kw">=</span> <span class="no">chr</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="no">start</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="no">end</span>, <span class="kw">ref</span> <span class="kw">=</span> <span class="no">ref</span>, <span class="kw">alt</span> <span class="kw">=</span> <span class="no">alt</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"avsnp147"</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)

<span class="co"># Annotate using multiple database</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.merge.html">annotation.merge</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">anno.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"cosmic81"</span>, <span class="st">"avsnp147"</span>), <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)

<span class="co"># Database configuration file</span>
<span class="no">database.cfg</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'config/databases.toml'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"anor"</span>)

<span class="co"># Get anno.name needed input cols</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/get.annotation.needcols.html">get.annotation.needcols</a></span>(<span class="st">'avsnp147'</span>)

<span class="co"># Annotate avinput format R data and file using ANNOVAR</span>
<span class="no">chr</span> <span class="kw">=</span> <span class="st">"chr1"</span>
<span class="no">start</span> <span class="kw">=</span> <span class="st">"123"</span>
<span class="no">end</span> <span class="kw">=</span> <span class="st">"123"</span>
<span class="no">ref</span> <span class="kw">=</span> <span class="st">"A"</span>
<span class="no">alt</span> <span class="kw">=</span> <span class="st">"C"</span>
<span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="no">chr</span>, <span class="no">start</span>, <span class="no">end</span>, <span class="no">ref</span>, <span class="no">alt</span>)
<span class="no">tmpfn</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>()
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(<span class="no">dat</span>, <span class="no">tmpfn</span>, <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">quote</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">sep</span> <span class="kw">=</span> <span class="st">"\t"</span>, <span class="kw">col.names</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="no">dat</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"perl_annovar_refGene"</span>, <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="no">annovar.dir</span>,
             <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">input.file</span> <span class="kw">=</span> <span class="no">tmpfn</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"perl_annovar_refGene"</span>, <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="no">annovar.dir</span>,
             <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)

<span class="co"># Annotate avinput format R data using anor and ANNOVAR</span>
<span class="co"># It will return a list contatin two data.table object that </span>
<span class="co"># one is anor annotation system and the another is ANNOVAR output </span>
<span class="no">x</span> <span class="kw">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.merge.html">annotation.merge</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">anno.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'avsnp147'</span>, <span class="st">'perl_annovar_refGene'</span>),
  <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="no">annovar.dir</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.merge.html">annotation.merge</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">anno.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'avsnp147'</span>, <span class="st">'1000g2015aug_all'</span>,
  <span class="st">'perl_annovar_refGene'</span>, <span class="st">'perl_annovar_ensGene'</span>), <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="no">annovar.dir</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)
<span class="co"># If use perl_annovar_merge as the anno.name, you can use annovar.anno.names to </span>
<span class="co"># run all original ANNOVAR supported annotation names, see http://annovar.openbioinformatics.org/en/latest/user-guide/download/</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.merge.html">annotation.merge</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">anno.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'avsnp147'</span>, <span class="st">'1000g2015aug_all'</span>,
  <span class="st">'perl_annovar_merge'</span>), <span class="kw">annovar.anno.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'refGene'</span>, <span class="st">'ensGene'</span>), <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="no">annovar.dir</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>)

<span class="co"># Annotate VCF file using ANNOVAR</span>
<span class="no">test.vcf.raw</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"demo/example.vcf"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"anor"</span>)
<span class="no">test.vcf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>()
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.copy</a></span>(<span class="no">test.vcf.raw</span>, <span class="no">test.vcf</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"perl_annovar_ensGene"</span>, <span class="kw">input.file</span> <span class="kw">=</span> <span class="no">test.vcf</span>,
             <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="no">annovar.dir</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>,
             <span class="kw">out</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(), <span class="kw">vcfinput</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co"># Annotate VCF file use VEP</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/vep.html">vep</a></span>(<span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"vep_all"</span>, <span class="kw">input.file</span> <span class="kw">=</span> <span class="no">test.vcf</span>,
             <span class="kw">out</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(), <span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co"># Annotate VCF file use vcfanno</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/vcfanno.html">vcfanno</a></span>(<span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"vcfanno_demo"</span>, <span class="kw">input.file</span> <span class="kw">=</span> <span class="no">test.vcf</span>,
                   <span class="kw">out</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(), <span class="kw">vcfanno</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"%s/vcfanno_linux64"</span>, <span class="no">vcfanno.dir</span>))

<span class="co"># Annotate VCF file use ANNOVAR and vcfanno</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.merge.html">annotation.merge</a></span>(<span class="kw">input.file</span> <span class="kw">=</span> <span class="no">test.vcf</span>, <span class="kw">anno.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"vcfanno_demo"</span>, <span class="st">"perl_annovar_ensGene"</span>),
  <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="no">annovar.dir</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">vcfanno</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"%s/vcfanno_linux64"</span>, <span class="no">vcfanno.dir</span>))

<span class="co"># Annotate gene using BioConductor database</span>
<span class="co"># The example below will use the org.Hs.eg.db to get the alias of TP53 and NSD2</span>
<span class="co"># It is more simple than the previous annotation API</span>
<span class="no">gene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"TP53"</span>, <span class="st">"NSD2"</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">gene</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"bioc_gene2alias"</span>)

<span class="co"># Do same things use AnnotationDbi</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">org.Hs.eg.db</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">AnnotationDbi</span>)
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span>(<span class="kw">keys</span> <span class="kw">=</span> <span class="no">gene</span>, <span class="kw">keytype</span> <span class="kw">=</span> <span class="st">"SYMBOL"</span>, <span class="kw">columns</span> <span class="kw">=</span> <span class="st">"ALIAS"</span>)</pre>
<p>There are still plenty of places to go for this tool. More annotation tools and annotation databases are being integrated. We hope this tool can provide some help for your variants annotation work.</p>
</div>
<div id="shiny-app" class="section level2">
<h2 class="hasAnchor">
<a href="#shiny-app" class="anchor"></a>Shiny App</h2>
<p>In the <a href="https://github.com/JhuangLab/BioInstaller">BoInstaller</a> bioshiny application, we developed several bioshiny plugins, such as <a href="http://annovar.openbioinformatics.org/en/latest/">ANNOVAR</a>, <a href="https://github.com/brentp/vcfanno">vcfanno</a>, <a href="https://github.com/PoisonAlien/maftools">maftools</a>, <a href="https://github.com/ytdai/gvmap">gvmap</a>, and so on.</p>
<pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">echo</span> <span class="st">'export BIO_SOFTWARES_DB_ACTIVE="~/.bioshiny/info.yaml" &gt;&gt; ~/.bashrc'</span>
<span class="kw">echo</span> <span class="st">'export BIOSHINY_CONFIG="~/.bioshiny/shiny.config.yaml" &gt;&gt; ~/.bashrc'</span>
<span class="kw">.</span> <span class="kw">~/.bashrc</span>

<span class="co"># Start the standalone Shiny application</span>
<span class="kw">wget</span> https://raw.githubusercontent.com/openbiox/bioshiny/master/bin/bioshiny_deps_r
<span class="kw">wget</span> https://raw.githubusercontent.com/openbiox/bioshiny/master/bin/bioshiny_deps_3rd
<span class="kw">wget</span> https://raw.githubusercontent.com/openbiox/bioshiny/master/bin/bioshiny_start
<span class="kw">chmod</span> a+x bioshiny_deps_r
<span class="kw">chmod</span> a+x bioshiny_deps_3rd
<span class="kw">chmod</span> a+x bioshiny_start
<span class="kw">./bioshiny_deps_r</span>
<span class="kw">./bioshiny_deps_3rd</span>

<span class="co"># Start Shiny application workers</span>
<span class="kw">Rscript</span> -e <span class="st">"bioshiny::set_shiny_workers(1)"</span>
<span class="kw">./bioshiny_start</span>

<span class="co"># or use yarn</span>
<span class="kw">yarn</span> global add bioshiny
<span class="kw">bioshiny_deps_r</span>
<span class="kw">bioshiny_deps_3rd</span>
<span class="kw">Rscript</span> -e <span class="st">"bioshiny::set_shiny_workers(1)"</span>
<span class="kw">bioshiny_start</span></code></pre>
</div>
<div id="how-to-contribute" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-contribute" class="anchor"></a>How to contribute?</h2>
<p>Please fork the <a href="https://github.com/JhuangLab/anor">GitHub anor repository</a>, modify it, and submit a pull request to us.</p>
</div>
<div id="maintainer" class="section level2">
<h2 class="hasAnchor">
<a href="#maintainer" class="anchor"></a>Maintainer</h2>
<p><a href="https://github.com/Miachol">Jianfeng Li</a></p>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor"></a>License</h2>
<p>R package:</p>
<p><a href="https://en.wikipedia.org/wiki/MIT_License">MIT</a></p>
<p>Related Other Resources:</p>
<p>Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/JhuangLab/anor/">https://​github.com/​JhuangLab/​anor/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/JhuangLab/anor/issues">https://​github.com/​JhuangLab/​anor/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Jianfeng Li <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2349-208X" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://circleci.com/gh/JhuangLab/anor/tree/master"><img src="https://img.shields.io/circleci/project/github/JhuangLab/anor/master.svg" alt="Build Status"></a></li>
<li><a href="https://cran.r-project.org/package=anor"><img src="http://www.r-pkg.org/badges/version/anor" alt="CRAN"></a></li>
<li><a href="http://www.r-pkg.org/pkg/anor"><img src="http://cranlogs.r-pkg.org/badges/anor?color=brightgreen" alt="Downloads"></a></li>
<li><a href="https://codecov.io/github/JhuangLab/anor"><img src="https://codecov.io/github/JhuangLab/anor/branch/master/graphs/badge.svg" alt="codecov"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="life">
  <p>anor was developed by <strong><a href="http://life2cloud.com">Jianfeng</a></strong>. </p>
</div>

<div class="author">
  <p>Developed by Jianfeng Li.</p>
  <p>Site built by <a href="http://pkgdown.r-lib.org">pkgdown</a>.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
